src/Microsoft.AspNet.Mvc.Core/ControllerActionInvokerProvider.cs

```diff
@@ -2,26 +2,43 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using Microsoft.AspNet.Mvc.ModelBinding;
 using Microsoft.Framework.DependencyInjection;
 
 namespace Microsoft.AspNet.Mvc
 {
     public class ControllerActionInvokerProvider : IActionInvokerProvider
     {
+        private readonly IControllerActionArgumentBinder _argumentBinder;
         private readonly IControllerFactory _controllerFactory;
-        private readonly IInputFormattersProvider _inputFormattersProvider;
         private readonly INestedProviderManager<FilterProviderContext> _filterProvider;
-        private readonly IControllerActionArgumentBinder _actionInvocationInfoProvider;
+        private readonly IInputFormattersProvider _inputFormatterProvider;
+        private readonly IInputFormatterSelector _inputFormatterSelector;
+        private readonly IModelBinderProvider _modelBinderProvider;
+        private readonly IModelMetadataProvider _modelMetadataProvider;
+        private readonly IModelValidatorProviderProvider _modelValidationProviderProvider;
+        private readonly IValueProviderFactoryProvider _valueProviderFactoryProvider;
 
-        public ControllerActionInvokerProvider(IControllerFactory controllerFactory,
-                                              IInputFormattersProvider inputFormattersProvider,
-                                              INestedProviderManager<FilterProviderContext> filterProvider,
-                                              IControllerActionArgumentBinder actionInvocationInfoProvider)
+        public ControllerActionInvokerProvider(
+            IControllerFactory controllerFactory,
+            IInputFormattersProvider inputFormatterProvider,
+            INestedProviderManager<FilterProviderContext> filterProvider,
+            IControllerActionArgumentBinder argumentBinder,
+            IModelMetadataProvider modelMetadataProvider,
+            IInputFormatterSelector inputFormatterSelector,
+            IModelBinderProvider modelBinderProvider,
+            IModelValidatorProviderProvider modelValidationProviderProvider,
+            IValueProviderFactoryProvider valueProviderFactoryProvider)
         {
             _controllerFactory = controllerFactory;
-            _inputFormattersProvider = inputFormattersProvider;
+            _inputFormatterProvider = inputFormatterProvider;
```

Comment:
Actually there's a bunch of these that are wrong. I'm going to rename the field and just leave it.